src/csvcubed/utils/sparql_handler/data_cube_inspector.py

```diff
@@ -207,6 +217,50 @@ def get_code_lists_and_cols(self, csv_url: str) -> CodelistsResult:
 
         return self._codelists_and_cols.get(csv_url, CodelistsResult([], 0))
 
+    def get_dataframe(self, csv_url: str) -> Tuple[pd.DataFrame, List[ValidationError]]:
+        """
+        Get the pandas dataframe for the csv url of the cube wishing to be loaded.
+        """
+        cols = self.get_column_component_info(csv_url)
+        dict_of_types = {}
+        for col in cols:
+            is_attribute_literal = (
+                col.column_type == EndUserColumnType.Attribute
+                and col.column_definition.value_url is None
+            )
+
+            if (
+                col.column_type == EndUserColumnType.Observations
+                or is_attribute_literal
+            ):
+                if col.column_definition.data_type is not None:
+                    col_data_type = col.column_definition.data_type.removeprefix(
+                        _XSD_BASE_URI
+                    )
+                else:
+                    raise ValueError(
+                        "Expected a defined datatype but got 'None' instead."
+                    )
+                if col_data_type in ACCEPTED_DATATYPE_MAPPING.keys():
+                    dict_of_types[
+                        col.column_definition.title
+                    ] = ACCEPTED_DATATYPE_MAPPING[col_data_type]
+                else:
+                    raise ValueError(
+                        f"Unhandled data type '{col.column_definition.data_type}'."
+                    )
+            else:
+                dict_of_types[col.column_definition.title] = "string"
+
+        absolute_csv_url = Path(
```

Comment:
Could you refactor this into a utils function which takes a URI and returns a file path? Ideally in the comment you could explain that the function works on both windows and lnux/unix.

There are a couple of other locations in the code base which do this same kind of thing. Could you tidy them up to use this new function too?